core/src/main/java/org/geysermc/geyser/GeyserImpl.java

```diff
@@ -393,6 +377,23 @@ private void startInstance() {
             }
         }
 
+        // Now that other platforms set the Bedrock port, also check the broadcast port
+        String broadcastPort = System.getProperty("geyserBroadcastPort", "");
+        if (!broadcastPort.isEmpty()) {
+            int parsedPort;
+            try {
+                parsedPort = Integer.parseInt(broadcastPort);
+                if (parsedPort < 1 || parsedPort > 65535) {
+                    throw new NumberFormatException("The broadcast port must be between 1 and 65535 inclusive!");
+                }
+            } catch (NumberFormatException e) {
+                logger.error(String.format("Invalid broadcast port: %s! Defaulting to configured port.", broadcastPort + " (" + e.getMessage() + ")"));
+                parsedPort = config.getBedrock().port();
+            }
+            config.getBedrock().setBroadcastPort(parsedPort);
+            logger.info("Broadcast port set from system property: " + parsedPort);
+        }
+
```

Comment:
looks to me like this: <https://github.com/GeyserMC/Geyser/blob/c685efa9b4c5fe2c6bc9cadc2ac09e52c083d6a6/core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java#L146-L149>

should be moved after all this property checking

Reply:
addressed in https://github.com/GeyserMC/Geyser/pull/4942/commits/b279ba7ca9d92b4ab93671e21263a282967ea99f